### PR TITLE
perf: add fragment pruning by _rowaddr filters in LanceScan

### PR DESCRIPTION
**Pull request by LuciferYang**
Closes https://github.com/lance-format/lance-spark/issues/117

### Summary

Added fragment-level pruning in the Spark read path based on `_rowaddr` filters — analogous to partition pruning in traditional Spark data sources. When a query includes `_rowaddr` predicates (e.g. from `DELETE`, `UPDATE`, or `MERGE` operations), `LanceScan.planInputPartitions()` now skips fragments that provably cannot match, eliminating unnecessary fragment opens, scan setup, and task scheduling.

### How it works

A new `RowAddressFilterAnalyzer` extracts target fragment IDs from pushed `_rowaddr` filters by decoding the fragment ID from the upper 32 bits of the row address (`fragment_id = (int)(rowAddr >>> 32)`). The analyzer supports `EqualTo`, `In`, `And` (intersection), and `Or` (union) filter types. `Not`, range filters, and non-numeric values conservatively fall back to no pruning — correctness is never compromised.

### Changes

| File | Change |
|------|--------|
| `RowAddressFilterAnalyzer.java` | New utility: recursively analyzes pushed filters to extract target fragment IDs from `_rowaddr` constraints |
| `LanceScan.java` | Added `pruneByRowAddrFilters()` helper that prunes splits before partition planning; passes `pushedFilters` from builder |
| `LanceScanBuilder.java` | Passes `pushedFilters` through to `LanceScan` constructor |
| `RowAddressFilterAnalyzerTest.java` | 31 unit tests covering all filter types, type coercion, nested boolean logic, edge cases |
| `BaseDeleteTableTest.java` | E2E test: `DELETE WHERE _rowaddr = X` only removes the target row across multi-fragment tables |

### Test plan

- [x] `RowAddressFilterAnalyzerTest` — 31 unit tests (filter analysis logic)
- [x] `BaseDeleteTableTest.testDeleteByRowAddrOnlyAffectsTargetFragment` — E2E integration test
- [x] `make install` — compiles cleanly across all modules
- [x] `make lint` — passes Checkstyle + Spotless

### CODE REVIEW

File: lance-spark-base_2.12/src/main/java/org/lance/spark/read/LanceScan.java [hunk 1]
```diff
@@ -123,6 +134,58 @@ public InputPartition[] planInputPartitions() {
...
+                  if (split.getFragments().size() > 1) {
+                    LOG.warn(
+                        "Split contains {} fragments;" + " sub-split pruning not implemented",
+                        split.getFragments().size());
+                  }
+                  return split.getFragments().stream().anyMatch(allowedIds::contains);
+                })
+            .collect(Collectors.toList());
+    if (pruned.size() < allSplits.size()) {
+      LOG.info(
```
Comment (hamersaw) [inline lance-spark-base_2.12/src/main/java/org/lance/spark/read/LanceScan.java]: Is this `debug` level?

**Comment by LuciferYang** [comment]
> @hamersaw [9ca0ffd](https://github.com/lance-format/lance-spark/commit/9ca0ffdd4484a0ef1d23f6e860ff379e4a4173d5) has addressed the comments.Could you take another look if you have the time? Thanks ~ and If the PR title and PR description need to be revised, please let me know.

I've updated the PR description with a new version.
